components/canvas/webgl_thread.rs

```diff
@@ -89,30 +88,130 @@ struct GLContextData {
     attributes: GLContextAttributes,
 }
 
+#[derive(Debug)]
 pub struct GLState {
     webgl_version: WebGLVersion,
     gl_version: GLVersion,
+    requested_flags: ContextAttributeFlags,
+    // This is the WebGL view of the color mask
+    // The GL view may be different: if the GL context supports alpha
+    // but the WebGL context doesn't, then color_write_mask.3 might be true
+    // but the GL color write mask is false.
+    color_write_mask: [bool; 4],
     clear_color: (f32, f32, f32, f32),
     scissor_test_enabled: bool,
+    // The WebGL view of the stencil write mask (see comment re `color_write_mask`)
     stencil_write_mask: (u32, u32),
+    stencil_test_enabled: bool,
     stencil_clear_value: i32,
+    // The WebGL view of the depth write mask (see comment re `color_write_mask`)
     depth_write_mask: bool,
+    depth_test_enabled: bool,
     depth_clear_value: f64,
+    // True when the default framebuffer is bound to DRAW_FRAMEBUFFER
+    drawing_to_default_framebuffer: bool,
     default_vao: gl::GLuint,
 }
 
+impl GLState {
+    // Are we faking having no alpha / depth / stencil?
+    fn fake_no_alpha(&self) -> bool {
+        self.drawing_to_default_framebuffer &
```

Comment:
A matter of style I think? I prefer `&` over `&&` in cases where the rhs is side-effect free, just because it generates straight-line code with no conditionals, which may play better with instruction pipelining / speculation?